controller_manager/src/controller_manager.cpp

```diff
@@ -294,15 +295,13 @@ bool ControllerManager::loadController(const std::string& name)
     try
     {
       // Trying loading the controller using all of our controller loaders. Exit once we've found the first valid loaded controller
-      std::list<ControllerLoaderInterfaceSharedPtr>::iterator it = controller_loaders_.begin();
-      while (!c && it != controller_loaders_.end())
+      for (auto const& it : controller_loaders_)
```

Comment:
Careful here. Read the above comment carefully. I agree that this part could be improved but I am unsure at the moment how best to achieve that. Your suggested change alters the algorithm

Reply:
```
for(const auto& it:controller_loaders_.begin()){
   if(c==controller_loaders_.end())
      break;
   for(const auto& cur_type:(it)->getDeclaredClass()){
      if(cur_type==type)
         c=(it)->createInstance(type);
      else
            c=it;
   }
}
```

Reply:
@bmagyar will this do?

Reply:
@bmagyar sorry I m utterly wrong did not read the code correctly.

Reply:
@bmagyar sir is my current implementation okay?

Reply:
Let's settle for a simpler solution, auto the iterator and leave the rest as-is.

`std::list<ControllerLoaderInterfaceSharedPtr>::iterator it = controller_loaders_.begin();` -> `auto it = controller_loaders_.begin();`